aiidalab_widgets_base/computers.py

```diff
@@ -450,161 +455,145 @@ def __private_key(self):
             return fname, content
         return None, None
 
-    @property
-    def hostname(self):
-        if not self._inp_computer_hostname.value.strip():  # check hostname
-            return None
-        return self._inp_computer_hostname.value
-
-    @hostname.setter
-    def hostname(self, hostname):
-        self._inp_computer_hostname.value = hostname
-
-    @property
-    def username(self):
-        """Loking for username in user's input and config file"""
-        if not self._inp_username.value.strip():  # if username provided by user
-            if self.hostname is not None:
-                config = parse_sshconfig(self.hostname)
-                if 'user' in config:  # if username is present in the config file
-                    return config['user']
-            else:
-                return None
-        return self._inp_username.value
-
-    @username.setter
-    def username(self, username):
-        self._inp_username.value = username
+    @validate('proxy_hostname')
+    def _validate_proxy_hostname(self, provided):
+        """Enable 'use proxy' widget if proxy hostname is provided."""
+        if provided['value']:
+            self.use_proxy = True
```

Comment:
Change of state should happen in the `observe` function, not in the validate function.

Reply:
ok